content/en/docs/wiki/How-to-contribute-large-features-into-Open-Y-(backporting,-etc).md

```diff
@@ -1,5 +1,69 @@
+These are our best practices for back-porting large features into YMCA Website Services and contributing code for others to use.
+
+## Summary
+
+* The YMCA Website Services core team is excited for you to contribute to the distribution for all to benefit from.
+* There's a lot that goes into back-porting your code -- details below.
+* These steps ensure that we are collaborating while continuing to support YMCA Website Services in a sustainable way.
+
+*Before getting started, please keep these notes in mind*
+
+* Back-porting requires a process called decoupling. This is where your developers remove any hard-coded variables or dependencies on any integration from your website code. This makes it so the feature you wish to contribute can work for anyone the broader Y movement. Meaning, *as an example* , anything that ties into Personify APIs will have to have those hooks replaced with Program Event Framework so that it could function with any CRM source that an YMCA Website Services site might be using.
+* This decoupled code will then need to be thoroughly tested to ensure it can function when not relying on any fee-based, or non-secure, technology or systems .
+* Once the decoupling is complete, the YMCA Website Services core team will need to review every line of code that goes before it goes into the distribution . This helps us ensure it won't break any of the other elements of YMCA Website Services, and that the code is 100% secure. As of 2019 we have 3 templates in YMCA Website Services that work across mobile, desktop and tablet breakpoints . We need to test new features/code across all these templates to make sure nothing breaks across hundreds of possible use-case scenarios ( *example* : adding a schedule block to a location landing page using the Carnation template when on the mobile breakpoint that is powered by ActiveNet through Program Event Framework...)
+* Once a feature is in YMCA Website Services, someone must pay to maintain that code . Does this code rely on any other Drupal modules to function? What if there's an update to that module? That update needs to be tested to ensure it's compatible with the now contributed code -- and if it breaks, we have to write new code to fix it. What if there's a security patch that involves the contributed code? We would then have to spend time applying the patch, etc.
+* Is your customization a new feature or a replacement of existing YMCA Website Services functionality/UX? We prefer you have A/B testing data demonstrating that your customization is a clear improvement over the current YMCA Website Services experience. There are many ways to run AB tests so please consult with the YMCA Website Services team on your hypothesis, method, and success criteria to ensure that the results are valid and reliable.
+
+*Steps/process for back-porting code into YMCA Website Services*
+
+1. List each customization/feature you want to contribute to YMCA Website Services
```

Comment:
0. It's best to develop your feature already decoupled, ready to be used as a module/project for not only your site or project but for other variety of Drupal or Javascript projects with a limited number of dependencies.  

@froboy

Reply:
@podarok how's this?

```suggestion
> Most problems have at least a generic component and can be approached in part through abstracted development.

We recommend beginning development with an eye toward [these "abstracted" solutions](https://guidebook.civicactions.com/en/latest/practice-areas/engineering/drupal/most-important-decision-in-developing-a-drupal-site-contributed-vs-custom-development/) - providing configuration instead of static templates, solving root causes instead of using local patches, using generic language instead of client specific. This will ensure that your features are easily contributed even before you begin this process.

1. List each customization/feature you want to contribute to YMCA Website Services
```